plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/preferences/SimplePreferenceStore.java

```diff
@@ -35,14 +36,31 @@ public abstract class SimplePreferenceStore extends AbstractPreferenceStore {
     private Map<String, String> defaultProperties;
     private boolean dirty = false;
 
-    public SimplePreferenceStore()
-    {
+    public interface Mutable {
```

Comment:
Maybe we should just show notification about preferences change?

Reply:
There is enough synchronization for SQL Templates without additional synchronization in SimplePreferenceStore. We can handle cases with other settings when we will work on it.